src/main/java/com/uid2/client/PublisherUid2Helper.java

```diff
@@ -0,0 +1,112 @@
+package com.uid2.client;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.time.Instant;
+import java.util.Arrays;
+
+
+public class PublisherUid2Helper {
+    public PublisherUid2Helper(String base64SecretKey) {
+        secretKey = InputUtil.base64ToByteArray(base64SecretKey);
+    }
+
+    public Envelope createEnvelope(IdentityInput identityInput) {
```

Comment:
Since we are using term `response` below, it would be great to create symmetry by introducing corresponding `request`. For example: `createTokenGenerateRequest()`.

Reply:
We're returning an Envelope instance though, not a Request. could do createEnvelopeForTokenGenerateRequest()?